tests/e2e/egress/egress_test.go

```diff
@@ -57,400 +48,132 @@ func TestE2EEgressConnectivity(t *testing.T) {
+			cmd:               []string{"curl", "-sSL", "-m", "10", "--fail-with-body", "https://httpbin.org/headers"},
+			applyEgressConfig: true,
+			expectError:       false,
 		},
 		{
 			name:               "connection to httpbin service is refused when NetworkPolicy is applied",
-			cmd:                []string{"curl", "-sSL", "-m", "10", "https://httpbin.org/headers"},
+			cmd:                []string{"curl", "-sSL", "-m", "10", "--fail-with-body", "https://httpbin.org/headers"},
 			applyNetworkPolicy: true,
 			// sidecar init fails when NP is applied. When uncommented, the test will pass despite confirming manually
 			// that connection is refused with NP
-			enableIstioInjection: true,
-			expectError:          true,
+			expectError: true,
 		},
 		{
-			name:                 "connection to httpbin service is OK when NetworkPolicy is applied and egress is configured",
-			cmd:                  []string{"curl", "-sSL", "-m", "10", "https://httpbin.org/headers"},
-			enableIstioInjection: true,
-			applyEgressConfig:    true,
-			applyNetworkPolicy:   true,
-			expectError:          false,
+			name:               "connection to httpbin service is OK when NetworkPolicy is applied and egress is configured",
+			cmd:                []string{"curl", "-sSL", "-m", "10", "--fail-with-body", "https://httpbin.org/headers"},
+			applyEgressConfig:  true,
+			applyNetworkPolicy: true,
+			expectError:        false,
 		},
 		{
-			name:                 "connection to kyma-project is refused when NetworkPolicy is applied and egress is configured",
-			cmd:                  []string{"curl", "-sSL", "-m", "10", "https://kyma-project.io"},
-			enableIstioInjection: true,
-			applyEgressConfig:    true,
-			applyNetworkPolicy:   true,
-			expectError:          true,
+			name:               "connection to kyma-project is refused when NetworkPolicy is applied and egress is configured",
+			cmd:                []string{"curl", "-sSL", "-m", "10", "--fail-with-body", "https://kyma-project.io"},
+			applyEgressConfig:  true,
+			applyNetworkPolicy: true,
+			expectError:        true,
 		},
 	}
 
 	for _, tt := range tc {
 		t.Run(tt.name, func(t *testing.T) {
-			ns := createTestNamespace(ctx, c, tt.enableIstioInjection, t)
+			runID := envconf.RandomName("e2e-egress", 16)
+			// instantiate a resources client
+			r, tErr := resources.New(cfg.Client().RESTConfig())
+			require.NoError(t, tErr)
+
+			// namespace creation
+			require.NoError(t, ns.CreateNamespace(ctx, t, runID, cfg,
+				ns.WithLabels(map[string]string{"istio-injection": "enabled"}),
+			))
 			if tt.applyEgressConfig {
-				err := applyEgressConfig(ctx, c, ns)
-				if err != nil {
-					t.Errorf("failed to create egress resources: %v", err)
-				}
+				t.Log("Applying egress config for the test: ", runID)
+				require.NoError(t, decoder.DecodeEachFile(ctx, fsys, "egress_config.yaml", decoder.CreateHandler(r), decoder.MutateNamespace(runID)))
```

Comment:
We should make sure that we are using the `decoder` package, or the `tests/e2e/e2e/helpers/infrastructure/yamlfile` everywhere.